### PR TITLE
[IMP] pos_self_order: update linting rules

### PR DESCRIPTION
**Pull request by vlst-odoo**
In this commit we update the lists of modules from `_eslintignore` and
`jsonconfig.json` such that they now contain all pos modules.

In addition, we also create script in the `tools` directory of pos that
returns a list of all the pos modules. This list is often very useful.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/tools/list_pos_modules.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import os
+import re
+
+def get_folders(path, regex):
```
Comment (caburj) [inline addons/point_of_sale/tools/list_pos_modules.py:4]: can you check if we can use `get_modules` instead? (`from odoo.modules.module import get_modules`)

Comment (vlst-odoo) [reply addons/point_of_sale/tools/list_pos_modules.py:4]: it looks like the import from odoo doesn't work in that directory

File: addons/web/tooling/_jsconfig.json [hunk 2]
```diff
@@ -25,29 +25,67 @@
             "@mail/*": ["addons/mail/static/src/*"],
             "@im_livechat/*": ["addons/im_livechat/static/src/*"],
             "@whatsapp/*": ["addons/whatsapp/static/src/*"],
+
             "@point_of_sale/*": ["addons/point_of_sale/static/src/*"],
+            "@pos_restaurant_stripe/*": ["pos_restaurant_stripe/static/src/*"],
+            "@pos_hr_restaurant/*": ["pos_hr_restaurant/static/src/*"],
+            "@pos_self_order/*": ["pos_self_order/static/src/*"],
```
Comment (caburj) [inline addons/web/tooling/_jsconfig.json:32]:
```suggestion
            "@pos_self_order/*": ["addons/pos_self_order/static/src/*"],
```

You forgot `addons` that's why it wasn't working after refresh.

Comment (vlst-odoo) [reply addons/web/tooling/_jsconfig.json:32]: i'll change it now

File: addons/web/tooling/_eslintignore [hunk 3]
```diff
@@ -217,20 +223,30 @@ addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js
 !addons/l10n_in_pos/**/*
 !addons/l10n_sa_pos
 !addons/l10n_sa_pos/**/*
+!addons/point_of_sale
+!addons/point_of_sale/**/*
```
Comment (caburj) [inline addons/web/tooling/_eslintignore:227]: duplicate. already listed above.

File: addons/web/tooling/_eslintignore [hunk 4]
```diff
@@ -239,26 +255,72 @@ addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js
...
 !addons/pos_six
 !addons/pos_six/**/*
 !addons/pos_stripe
 !addons/pos_stripe/**/*
+!addons/spreadsheet_dashboard_pos_hr
+!addons/spreadsheet_dashboard_pos_hr/**/*
 
 # Whitelist enterprise pos modules
+!l10n_be_reports_post_wizard
+!l10n_be_reports_post_wizard/**/*
```
Comment (caburj) [inline addons/web/tooling/_eslintignore:283]: Why is this here?

File: addons/web/tooling/_jsconfig.json [hunk 5]
```diff
@@ -25,29 +25,68 @@
...
+            "@pos_l10n_se/*": ["pos_l10n_se/static/src/*"],
+            "@pos_online_payment_self_order_preparation_display/*": ["pos_online_payment_self_order_preparation_display/static/src/*"],
+            "@pos_order_tracking_display/*": ["pos_order_tracking_display/static/src/*"],
+            "@pos_preparation_display/*": ["pos_preparation_display/static/src/*"],
+            "@pos_restaurant_appointment/*": ["pos_restaurant_appointment/static/src/*"],
+            "@pos_restaurant_preparation_display/*": ["pos_restaurant_preparation_display/static/src/*"],
+            "@pos_sale_stock_renting/*": ["pos_sale_stock_renting/static/src/*"],
+            "@pos_self_order_preparation_display/*": ["pos_self_order_preparation_display/static/src/*"],
+            "@pos_settle_due/*": ["pos_settle_due/static/src/*"],
+            "@whatsapp_pos/*": ["whatsapp_pos/static/src/*"],
```
Comment (caburj) [inline addons/web/tooling/_jsconfig.json:87]: Screenshot 2023-11-28 at 14 59 56

It is originally with `addons`. Did they make a mistake when they declared it here?

Comment (vlst-odoo) [reply addons/web/tooling/_jsconfig.json:87]: yes

**Review by caburj — APPROVED** [review]
robodoo r+
